### PR TITLE
feat: style operations in search item component

### PR DESCRIPTION
**Pull request by ohorbachevskyi**
## What/Why/How?
This PR adds additional styling into search result item of ref docs operation.

## Screenshots (optional)
Screenshot 2022-04-21 at 15 44 24
Screenshot 2022-04-21 at 15 21 34

## Check yourself

- [ ] Code is linted
- [ ] Tested
- [ ] All new/updated code is covered with tests

## Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### CODE REVIEW

File: src/Search/SearchItem.tsx [hunk 1]
```diff
@@ -25,12 +26,17 @@ export function SearchItem({ item }: SearchItemProps) {
       ref.current.focus();
     }
   }, [item.active]);
-
   return (
-    <SearchLink to={item.url} tabIndex={0} innerRef={ref} data-component-name="Search/SearchItem">
-      <Title>{item.title}</Title>
+    <SearchLink to={item.url} tabIndex={0} innerRef={ref}>
+      <Title>
+        {item.httpVerb && <Badge type={item.httpVerb}>{item.httpVerb}</Badge>}
```
Comment (ozapotichnyi) [inline src/Search/SearchItem.tsx:32]:
```suggestion
        {item.httpVerb ? <Badge type={item.httpVerb}>{item.httpVerb}</Badge> : null}
```

File: src/Search/SearchItem.tsx [hunk 2]
```diff
@@ -25,12 +26,17 @@ export function SearchItem({ item }: SearchItemProps) {
...
+        {item.title}
+      </Title>
       <Description>{highlight(item.chunks)}</Description>
-      <Path>{item.path.join(' > ')}</Path>
+      <Path>
+        {item.place && (item.place + ' > ')}
```
Comment (ozapotichnyi) [inline src/Search/SearchItem.tsx:37]: What's place?
Can we push it in path?

Comment (ozapotichnyi) [reply src/Search/SearchItem.tsx:37]: Or concat before?

**Review by ozapotichnyi — CHANGES_REQUESTED** [review]

**Review by slavikbez — APPROVED** [review]

File: src/Search/SearchItem.tsx [hunk 3]
```diff
@@ -59,14 +124,22 @@ const Title = styled.div`
...
   text-overflow: ellipsis;
+  
+  &:first-child {
+    padding-top: 0;
+  }
+`;
+
+const Path = styled.div`
+  font-size: 12px;
+  line-height: 22px;
```
Comment (slavikbez) [inline src/Search/SearchItem.tsx:144]: I believe there are css variables from `font-size` and `line-height` properties. Try to use them

File: src/Search/SearchItem.tsx [hunk 4]
```diff
@@ -27,10 +40,38 @@ export function SearchItem({ item }: SearchItemProps) {
...
   return (
-    <SearchLink to={item.url} tabIndex={0} innerRef={ref} data-component-name="Search/SearchItem">
-      <Title>{item.title}</Title>
-      <Description>{highlight(item.chunks)}</Description>
-      <Path>{item.path.join(' > ')}</Path>
+    <SearchLink to={item.url} tabIndex={0} innerRef={ref}>
+      <Operaition>
+        {item.httpVerb ? <OperationBadge type={item.httpVerb}>{item.httpVerb}</OperationBadge> : null}
+        {highlight(item.pathName)}
+      </Operaition>
```
Comment (ozapotichnyi) [inline src/Search/SearchItem.tsx:47]: I'd hide operation for regular text results (Mardkdown):
```suggestion
      {item.httpVerb ? <Operaition>
          <OperationBadge type={item.httpVerb}>{item.httpVerb}</OperationBadge> 
          {highlight(item.pathName)}
        </Operaition>
      : null}
```

File: src/Search/SearchItem.tsx [hunk 5]
```diff
@@ -27,10 +40,38 @@ export function SearchItem({ item }: SearchItemProps) {
...
+            return <>
+              <Place>
+                {path}{highlight(param.name)} <br/>
+                {highlight(param.description)}
+              </Place>
+            </>
+            })
+          }
+          {moreItems ? <MoreText>and {moreItems} more...</MoreText> : null}
+        </Parameters>
```
Comment (ozapotichnyi) [inline src/Search/SearchItem.tsx:69]: Let's extract it into separate component
```suggestion
        <Parameters parameters={item.parameters} />
```

**Review by ozapotichnyi — CHANGES_REQUESTED** [review]

File: src/Search/Parameters.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+const MAX_ITEMS = 2;
+
+export function Parameters({ parameters = []}: ParametersProps) {
+  let moreItems = 0;
+
+  if (parameters.length > MAX_ITEMS) {
+    moreItems = parameters.length - MAX_ITEMS;
+    parameters.length = MAX_ITEMS;
+  }
```
Comment (ozapotichnyi) [inline src/Search/Parameters.tsx:18]: const moreItems = Math.max(0, parameters.length - MAX_ITEMS)

Comment (ozapotichnyi) [inline src/Search/Parameters.tsx:18]:
```suggestion
  const moreItems = Math.max(0, parameters.length - MAX_ITEMS)
```

**Review by ozapotichnyi — APPROVED** [review]

**Pull request merged**